tabular/src/autogluon/tabular/models/pecos/run.py

```diff
@@ -0,0 +1,124 @@
+# Run PECOS with AutoGluon. Testing file provided for convenience
+from autogluon.tabular import TabularDataset
+from pecos_model import PecosModel
+import pandas as pd
+
+train_data = TabularDataset('https://autogluon.s3.amazonaws.com/datasets/Inc/train.csv')  # can be local CSV file as well, returns Pandas DataFrame
+test_data = TabularDataset('https://autogluon.s3.amazonaws.com/datasets/Inc/test.csv')  # another Pandas DataFrame
+label = 'class'  # specifies which column do we want to predict
+
+print(train_data.head(5))
+
+# Clean data
+
+# Separate features and labels
+X = train_data.drop(columns=[label])
+y = train_data[label]
+X_test = test_data.drop(columns=[label])
+y_test = test_data[label]
+
+from autogluon.core.data import LabelCleaner
+from autogluon.core.utils import infer_problem_type
+# Construct a LabelCleaner to neatly convert labels to float/integers during model training/inference, can also use to inverse_transform back to original.
+problem_type = infer_problem_type(y=y)  # Infer problem type (or else specify directly)
+label_cleaner = LabelCleaner.construct(problem_type=problem_type, y=y)
+y_clean = label_cleaner.transform(y)
+
+print(f'Labels cleaned: {label_cleaner.inv_map}')
+print(f'inferred problem type as: {problem_type}')
+print('Cleaned label values:')
+print(y_clean.head(5))
+
+
+# Clean features (i.e. strings)
+
+from autogluon.common.utils.log_utils import set_logger_verbosity
+from autogluon.features.generators import AutoMLPipelineFeatureGenerator
+set_logger_verbosity(2)  # Set logger so more detailed logging is shown for tutorial
+
+feature_generator = AutoMLPipelineFeatureGenerator()
+X_clean = feature_generator.fit_transform(X)
+
+print(X_clean.head(5))
+
+# Fit model
+
+cat_features = ['workclass', 'education', 'marital-status', 'occupation', 'relationship', 'race', 'native-country']
+text_features = None
+num_features = ['age', 'fnlwgt', 'education-num', 'sex', 'capital-gain', 'capital-loss', 'hours-per-week']
```

Comment:
this will automatically be stored in `self._feature_metadata` in PecosModel during the call to `_fit`. No need to define it here.

Reply:
Fixed!